rust/build_assert.rs

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Build-time assertion.
+//!
+//! In Rust it is very common to use assertions extensively and defensively.
+//! Sometimes we know a condition could be checked statically, but it could not
+//! be enforced in Rust (e.g. perform some checks in const functions, but those
+//! functions could still be called in the runtime).
+//!
+//! This crate provides a method `build_error`, which will panic in
+//! compile-time if executed in const context, and will cause a build error
+//! if not executed in compile time and optimizer does not optimise away the
+//! call.
+
+#![no_std]
+#![feature(const_panic, core_panic)]
+
+/// Panics if executed in const context, or triggers a build error if not.
+#[inline(never)]
+#[no_mangle]
+#[track_caller]
+pub const fn build_error(msg: &'static str) {
+    // Could also be panic!(msg) to avoid using unstable feature `core_panic`,
+    // but it is not allowed in Rust 2021, while panic!("{}", msg) could not
```

Comment:
```suggestion
    // Could also be `panic!(msg)` to avoid using unstable feature `core_panic`,
    // but it is not allowed in Rust 2021, while `panic!("{}", msg)` could not.
```